p4-16/psa/examples/psa-example-hello-world.p4

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2019 Cisco Systems, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <core.p4>
+#include "../psa.p4"
+
+
+typedef bit<48>  EthernetAddress;
+
+header ethernet_t {
+    EthernetAddress dstAddr;
+    EthernetAddress srcAddr;
+    bit<16>         etherType;
+}
+
+header ipv4_t {
+    bit<4>  version;
+    bit<4>  ihl;
+    bit<8>  diffserv;
+    bit<16> totalLen;
+    bit<16> identification;
+    bit<3>  flags;
+    bit<13> fragOffset;
+    bit<8>  ttl;
+    bit<8>  protocol;
+    bit<16> hdrChecksum;
+    bit<32> srcAddr;
+    bit<32> dstAddr;
+}
+
+struct empty_metadata_t {
+}
+
+struct fwd_metadata_t {
+}
+
+struct metadata {
+    fwd_metadata_t fwd_metadata;
+}
+
+struct headers {
+    ethernet_t       ethernet;
+    ipv4_t           ipv4;
+}
+
+parser IngressParserImpl(packet_in buffer,
+                         out headers parsed_hdr,
+                         inout metadata user_meta,
+                         in psa_ingress_parser_input_metadata_t istd,
+                         in empty_metadata_t resubmit_meta,
+                         in empty_metadata_t recirculate_meta)
+{
+    state start {
+        transition parse_ethernet;
+    }
+    state parse_ethernet {
+        buffer.extract(parsed_hdr.ethernet);
+        transition select(parsed_hdr.ethernet.etherType) {
+            0x0800: parse_ipv4;
+            default: accept;
+        }
+    }
+    state parse_ipv4 {
+        buffer.extract(parsed_hdr.ipv4);
+        transition accept;
+    }
+}
+
+control ingress(inout headers hdr,
+                inout metadata user_meta,
+                in    psa_ingress_input_metadata_t  istd,
+                inout psa_ingress_output_metadata_t ostd)
+{
+    apply {
+        if (hdr.ipv4.isValid()) {
+            // Direct IPv4 packets out of ports 0 through 3, based
+            // upon the 2 least significant bits of the IPv4
+            // destination address.
+            send_to_port(ostd,
+                (PortId_t) ((PortIdUint_t) hdr.ipv4.dstAddr[1:0]));
```

Comment:
The latest master fails to parse this line? 
```
hanwang-lt1:build hanwang$ ./p4c-bm2-psa --std p4-16 psa-example-hello-world.p4 -Tpass_manager:1
psa-example-hello-world.p4(93):syntax error, unexpected IDENTIFIER "hdr"
                (PortId_t) ((PortIdUint_t) hdr
                                           ^^^
error: 1 errors encountered, aborting compilation
```

Reply:
Do you see anything wrong with the line of code?  If not, it seems like it must be a bug in p4c-bm2-psa to me.

Reply:
The latest p4test parses it without errors.

Reply:
I don't see anything wrong in the source. must be a bug in the frontend.

Reply:
ah, it is because I do not have the updated psa.p4.